examples/mapping_functions.py

```diff
@@ -0,0 +1,445 @@
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+from loguru import logger
+from numpy.typing import NDArray
+from scipy.sparse import csr_matrix, dia_matrix
+from xmipy import XmiWrapper
+
+from imod_coupler.utils import create_mapping
+
+
+def unique_list(input):
+    output = np.unique(np.array(input))
+    return list(output)
+
+
+# mapping different types of exchanges within DFLOWMETMOD driver
+def mapping_active_MF_DFLOW1D(workdir, dflow1d_lookup, array: Optional[NDArray] = None):
+    # function creates dictionary with mapping tables for mapping MF <-> dflow1d
+    # mapping includes active MF coupling:
+    #   1 MF RIV 1                      -> DFLOW-FM 1D flux
+    #   2 DFLOW FM 1D (correction)flux  -> MF RIV 1
+    #   3 DFLOW FM 1D stage             -> MF RIV 1
+    map_active_mod_dflow1d: Dict[str, csr_matrix] = {}
+    mask_active_mod_dflow1d: Dict[str, NDArray[Any]] = {}
+
+    # MF RIV 1 -> DFLOW 1D (flux)
+    mapping_file = workdir / "MFRIVTODFM1D_Q.DMM"
+    if mapping_file.is_file():
+        table_active_mfriv2dflow1d: NDArray[np.single] = np.loadtxt(
+            mapping_file, dtype=np.single, ndmin=2, skiprows=1
+        )
+        mf_idx = table_active_mfriv2dflow1d[:, 2].astype(int) - 1
+        dflow_idx = np.array(
+            [
```

Comment:
I think here you don't need to go back to the ii index. You  might be able to fill this list as:

   [dflow1d_lookup[row[0], row[1]  for row in table_active_mfriv2dflow1d)   ]